templates/ecs.yaml

```diff
@@ -0,0 +1,317 @@
+
+  # Security group for the ALB that goes in front
+  # Should only allow port 80 from the world for now
+  EcsAlbSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Load Balancer for ECS security group
+      VpcId: !Ref 'VpcId'
+  EcsAlbSecurityGroupHTTPinbound:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsAlbSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 80
+      ToPort: 80
+      CidrIp: 0.0.0.0/0
+
+  CloudwatchLogsGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Join ['-', [ECSLogGroup, !Ref 'AWS::StackName']]
+      RetentionInDays: 14
+
+  # The 'Family' property is key here, it tells ECS to increment the version rather
+  # than create a new Task Definition
+  taskdefinition:
+    Type: AWS::ECS::TaskDefinition
+    Properties:
+      Family: !Ref 'AWS::StackName'
+      ExecutionRoleArn: !GetAtt ['IAtlasIAMExecutionRole', 'Arn']
+      ContainerDefinitions:
+      - Name: iatlas-api
+        Cpu: 100
+        Essential: true
+        Image: !Join [':', ['registry.gitlab.com/cri-iatlas/iatlas-api', !Ref 'DockerImageTag']]
+        RepositoryCredentials:
+          CredentialsParameter: !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:staging/gitlab-registry-credentials'
+        Memory: 300
+        LogConfiguration:
+          LogDriver: awslogs
+          Options:
+            awslogs-group: !Ref 'CloudwatchLogsGroup'
+            awslogs-region: !Ref 'AWS::Region'
+            awslogs-stream-prefix: iatlas-api
+        PortMappings:
+        - ContainerPort: 80
+        Environment:
+          - Name: FLASK_ENV
+            Value: 'production'
+        Secrets:
+          - Name: POSTGRES_DB
+            ValueFrom: !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:iatlas_api_creds_encrypted:dbname::'
+          - Name: POSTGRES_HOST
+            ValueFrom: !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:iatlas_api_creds_encrypted:host::'
+          - Name: POSTGRES_USER
+            ValueFrom: !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:iatlas_api_creds_encrypted:username::'
+          - Name: POSTGRES_PASSWORD
+            ValueFrom: !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:iatlas_api_creds_encrypted:password::'
+  ECSALB:
+    Type: AWS::ElasticLoadBalancingV2::LoadBalancer
+    Properties:
+      Name: ECSALB
+      Scheme: internet-facing
+      LoadBalancerAttributes:
+      - Key: idle_timeout.timeout_seconds
+        Value: '30'
+      Subnets: !Ref 'PublicSubnetId'
+      SecurityGroups: [!Ref 'EcsAlbSecurityGroup']
+  ALBListener:
+    Type: AWS::ElasticLoadBalancingV2::Listener
+    DependsOn: ECSServiceRole
+    Properties:
+      DefaultActions:
+      - Type: forward
+        TargetGroupArn: !Ref 'ECSTG'
+      LoadBalancerArn: !Ref 'ECSALB'
+      Port: 80
+      Protocol: HTTP
```

Comment:
i assume HTTPS will come later?

Reply:
Khai is correct to point out the requirement here too